DesignSafe-CI/portal
# Bug/DES-2778: Handle number validation in App Form

## Overview: ##
Form input is string, but the field validation is number based. The necessary conversion is needed in zod validation
## PR Status: ##


## Related Jira tickets: ##

* [DES-2778](https://tacc-main.atlassian.net/browse/DES-2778)

## Summary of Changes: ##
1. Use a string to number converter
2. Apply the preprocessor to all zod validations for numbers.

## Testing Steps: ##
1. Go to any app (example: https://designsafe.dev/rw/workspace/openfoam#!/) 
2. go to the configuration step and modify various numbers.
3. Submit job to check that data is accepted.

## UI Photos:

## Notes: ##
